File: app/src/main/java/com/kml/aLoginScreen/DbLogin.java
package com.kml.aLoginScreen;

import android.os.AsyncTask;
import android.util.Log;

import com.kml.aGlobalUses.ExternalDbHelper;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;

public class DbLogin extends ExternalDbHelper
{
    private HttpURLConnection httpConnection;
    private String result;
    private final String address = "http://sobos.ssd-linuxpl.com/login.php";
    private String login;
    private String password;

    public DbLogin(String login, String password)
    {
        this.login = login;
        this.password = password;
    }

    @Override
    public void run()
    {
        httpConnection = setConnection(address);
        sendData();
        result = readResult(httpConnection);
    }


    private void sendData()
    {
        try {
            OutputStream outStream = httpConnection.getOutputStream();
            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"));
            String dataToSend = URLEncoder.encode("user", "UTF-8") + "=" + URLEncoder.encode(login, "UTF-8")
                    + "&&" + URLEncoder.encode("pass", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8");

            writer.write(dataToSend);
            writer.flush();
            writer.close();
        }
        catch (IOException e)
        {
            Log.d(IO_EXCEPTION_TAG, "sendData: "+e.getMessage());
        }
    }

    public String getResult()
    {
        try {
            join();
        } catch (InterruptedException e) {
            Log.d(IO_EXCEPTION_TAG, "getResult: "+e.getMessage());
        }
        return result;
    }
}

File: app/src/main/java/com/kml/data/externalDbOperations/DbLogin.kt
package com.kml.data.externalDbOperations

import android.util.Log
import java.io.BufferedWriter
import java.io.IOException
import java.io.OutputStreamWriter
import java.net.HttpURLConnection
import java.net.URLEncoder

class DbLogin(private val login: String, private val password: String) : ExternalDbHelper() {
    private val fileName = "login.php"
    private var httpConnection: HttpURLConnection? = null
    var result: String = ""
    get() { join(); return field}
    private val address: String = BASE_URL + fileName


    override fun run() {
        httpConnection = setConnection(address)
        sendData()
        result = readResult(httpConnection!!)
    }

    private fun sendData() {
        try {
            val outStream = httpConnection?.outputStream
            val writer = BufferedWriter(OutputStreamWriter(outStream, "UTF-8"))
            val dataToSend = (URLEncoder.encode("user", "UTF-8") + "=" + URLEncoder.encode(login, "UTF-8")
                    + "&&" + URLEncoder.encode("pass", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8"))
            writer.write(dataToSend)
            writer.flush()
            writer.close()
        } catch (e: IOException) {
            Log.d(IO_EXCEPTION_TAG, "sendData: " + e.message)
        }
    }
}